core/dbt/parser/schemas.py

```diff
@@ -326,19 +341,35 @@ def _parse_generic_test(
             raw_sql=builder.build_raw_sql(),
             column_name=column_name,
             test_metadata=metadata,
+            file_key_name=file_key_name,
         )
-        self.render_test_update(node, config, builder)
+        self.render_test_update(node, config, builder, schema_file_id)
 
         return node
 
+    def store_env_vars(self, target, schema_file_id, env_vars):
+        self.manifest.env_vars.update(env_vars)
+        if schema_file_id in self.manifest.files:
+            schema_file = self.manifest.files[schema_file_id]
+            if isinstance(target, UnpatchedSourceDefinition):
+                search_name = target.source.name
+                yaml_key = target.source.yaml_key
+                if '.' in search_name:  # source file definitions
+                    (search_name, _) = search_name.split('.')
+            else:
+                search_name = target.search_name
```

Comment:
I was trying something pretty wacky, and managed to break this:
```
version: 2
models:
  - name: model_a
    columns:
      - name: id
        tests:
          - unique:
              enabled: "{{ env_var('ENABLED', True) }}"
```
```
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/manifest.py", line 439, in parse_project
    parser.parse_file(block, dct=dct)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 507, in parse_file
    self.parse_tests(test_block)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 483, in parse_tests
    self.parse_column_tests(block, column)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 207, in parse_column_tests
    self.parse_test(block, test, column)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 479, in parse_test
    self.parse_node(block)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 417, in parse_node
    node = self._parse_generic_test(
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 295, in _parse_generic_test
    self.store_env_vars(target, schema_file_id, self.schema_yaml_vars.env_vars)
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/schemas.py", line 360, in store_env_vars
    search_name = target.search_name
```

Reply:
Will make a test case for it.

Reply:
This should be fixed. I also added throwing an exception for when there is no node to match a patch, because part of the test case was not doing what it was supposed to.